app/src/main/java/com/eveningoutpost/dexdrip/G5Model/Ob1G5StateMachine.java

```diff
@@ -610,7 +610,9 @@ public static boolean doGetData(Ob1G5CollectionService parent, RxBleConnection c
                                 UserError.Log.ueh(TAG, "Session Start failed info: " + JoH.dateTimeText(session_start.getSessionStart()) + " " + JoH.dateTimeText(session_start.getRequestedStart()) + " " + JoH.dateTimeText(session_start.getTransmitterTime()));
                                 if (session_start.isFubar()) {
                                     final long tk = DexTimeKeeper.getDexTime(getTransmitterID(), tsl());
-                                    if (tk > 0) {
+                                    if (tk > 0 & FirmwareCapability.isTransmitterRawIncapable(getTransmitterID())) { // Firefly, which cannot be hard reset
```

Comment:
I suggest changing this to: 
```java
                                    if (tk > 0) {
                                        if (FirmwareCapability.isTransmitterRawIncapable(getTransmitterID())) { // Firefly, which cannot be hard reset
                                            UserError.Log.e(TAG, "Unusual session start failure");
                                        else {
                                            DexResetHelper.offer("Unusual session start failure, is transmitter crashed? Try Hard Reset?");
                                        }
                                    } else {
```

But this is just a minor suggestion improving code readability.

Reply:
Makes sense.  Thanks.
Done